### ISSUE TITLE
[Calendar] Improve screenreader experience; only read visible text

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#2992]
### Provide a general summary of the issue here

Currently in the `readonly` variant days are not available for screen reader users.

Please check comment from Kasia: 
https://github.com/sl-design-system/components/issues/2568#issuecomment-3872412863

We need to make some more research before starting the development part.

### 👤 Your name

SLDS Team

### 🧢 Your Product/Team

SLDS

**Comment by a11ymiko** [comment]
After consulting with Katarzyna and blind users conclusion for that issue is not to change `tabindex` placement.
Tab order should remain as it is now, so only "month", "year", "previous month" and "next month" buttons are focusable. 
There would be `axe-core` Violation "Scrollable element focusable" but there this rule is not mandatory, as we have an alternative to scroll: arrow buttons that go to previous and next month and are fully keyboard accessible.
An exception for rule `scrollable-region-focusable` can be made on that story.

The only small improvement could be to hide both invisible `<table role="grid">` with previous and next month using `aria-hidden`. So when `<sl-month-view>` receives `aria-hidden='true'` also `<table role='grid'>` should be given `aria-hidden='true'`.
That would fix the issue, that is presented only on VoiceOver that announces each table as eg. "Days of January 2026, Days of February 2026, Days of March 2026, group" instead "Days of February 2026, group". 

**Current behavior:** 

https://github.com/user-attachments/assets/6c7e5307-3892-49bd-b7ca-cf32fc5d5426

<img width="393" height="470" alt="Image" src="https://github.com/user-attachments/assets/81e85ef9-7a3d-49bb-bba9-c26e9ea846f9" />

No `<table role='grid'>` has `aria-hidden='true'`

**Proposed solution**

https://github.com/user-attachments/assets/93282e2a-f574-4d2e-8ca8-cf6369ac1e31

<img width="441" height="444" alt="Image" src="https://github.com/user-attachments/assets/c58c8b13-f121-4411-9428-ca87c25e3e24" />

Each `<table role='grid'>` inside `<sl-month-view>` that has `ria-hidden='true'` also has `aria-hidden='true'`.
